atef/procedure.py

```diff
@@ -1,13 +1,52 @@
 from __future__ import annotations
 
+import asyncio
 import dataclasses
 import datetime
+import json
 import pathlib
 from dataclasses import dataclass, field
-from typing import Any, Dict, Optional, Sequence, Union
+from typing import (Any, Dict, Generator, Literal, Optional, Sequence, Union,
+                    cast)
+
+import apischema
+import databroker
+import yaml
+from bluesky import RunEngine
+
+from atef.cache import DataCache
+from atef.check import Result, incomplete
+from atef.config import (ConfigurationFile, PreparedFile,
+                         _summarize_result_severity, run_passive_step)
+from atef.enums import GroupResultMode, Severity
+from atef.type_hints import AnyPath
+from atef.yaml_support import init_yaml_support
 
 from . import serialization
 
+run_engine_singleton = None
+db_singleton = None
+
+
+def incomplete_result():
```

Comment:
This got defined here and in `atef.check`, it looks like?

Reply:
If I'm understanding my past self, I think I define the function here so I can pass into the dataclass default_factory, which doesn't like lambdas.  The shortcut `incomplete` was defined in `atef.check`, and just passed through here.  

I suppose I could redefine the helper function in each module I use it, maybe I took DRY a little too far.

Reply:
Yeah, the "no lambda" requirement makes sense. I was pointing at this:

```
$ git grep 'def incomplete_result'
atef/config.py:def incomplete_result():
atef/procedure.py:def incomplete_result():
```

But we've already spent more time than it's worth for removing 2 lines of code, so... yeah